lib/Reader/readerir.cpp

```diff
@@ -545,6 +545,16 @@ void GenIR::readerPostPass(bool IsImportOnly) {
     }
   }
 
+  // Crossgen in ReadyToRun mode records structs that method compilation
+  // depends on via getClassSize calls. Therefore, we shouldn't cache type info
+  // across methods.
+  if (JitContext->Flags & CORJIT_FLG_READYTORUN) {
```

Comment:
I think it's only safe to clear out the caches if we're at the "top-level" context. Presumably re-entrant jit requests are unlikely in a prejit scenario so this is probably an academic point, but can you add an assert anyways?